src/main/java/com/exasol/errorreporting/Quoter.java

```diff
@@ -14,21 +14,36 @@ private Quoter() {
 
     /**
      * Return a quoted string representation of the objectToQuote passed.
-     * 
-     * @param object object to quote
+     *
+     * @param object  object to quote
+     * @param quoting quoting style to be used
      * @return quoted object
      */
-    static String quoteObject(final Object object) {
+    static String quoteObject(final Object object, final Quoting quoting) {
         if (object == null) {
             return "<null>";
-        } else if (object instanceof String || object instanceof Character || object instanceof java.nio.file.Path
-                || object instanceof java.io.File || object instanceof java.net.URL || object instanceof java.net.URI) {
-            return "'" + object + "'";
         } else if (object instanceof List) {
```

Comment:
You could also use `Collection` or `Iterable` here, then we can handle all types of collections.

Reply:
Good idea. Done.